.github/workflows/github-pages.yml

```diff
@@ -0,0 +1,29 @@
+# GitHub Action for Jekyll
+#
+# Runs jekyll build on master and pushes output to gh-pages branch.
+# (Then GitHub deploys that static content.)
+
+name: Build and deploy Jekyll site to GitHub Pages
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  github-pages:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+          restoreKeys: |
+            ${{ runner.os }}-gems-
+      - uses: helaili/jekyll-action@2.0.4
+        env:
+          JEKYLL_PAT: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Would this require the token be set in the repo settings? Or is this auto-populated by GitHub?

One thing we'd have to be careful of is making sure that token is (a) somewhat limited (e.g. not a personal token for me, which would be able to reach a large swath of my stuff), and (b) not able to be gotten into in PR builds.

Reply:
Wow, this led me down a bit of a rabbit hole. (Learned a lot about Actions along the way)

GitHub does auto-populate the [GITHUB_TOKEN](https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#using-the-github_token-in-a-workflow) secret. However, the docs for [helaili/jekyll-actkon](https://github.com/marketplace/actions/jekyll-actions#known-limitation) basically say, "this might work or it might not, you can set up an extra personal access token instead.". A little disappointing since this is the Action Jekyll recommends [on their own site](https://jekyllrb.com/docs/continuous-integration/github-actions/). It seemed preferable (& more secure) to me to use the one GitHub provided, so I started experimenting a little. I found a different [GitHub Pages Action](https://github.com/marketplace/actions/github-pages-action) that's much more popular/well-maintained, and supports anything that generates static content, with well-documented support for `GITHUB_TOKEN` or a variety of other auth methods. So I'd propose we use that one instead, and I've updated this PR accordingly.

I tested all of this to confirm it actually works and to work out the kinks: https://github.com/mkasberg/jekyll-action-test

Since we're using the automatically generated `GITHUB_TOKEN`, it _is_ limited to this repository. And it's configured to only trigger on merges to the master branch - we wouldn't be running code from a PR (I don't think GitHub allows PR code to run before it's merged anyway).

Reply:
Alternatively, if you'd prefer to avoid all this and stick with the GitHub-built Jekyll deployment to GitHub pages, that's reasonable. LMK your thoughts.

Reply:
It does look like things should be protected enough, and it won't allow PRs to push to the main site, so I'm in favor of this change. Definitely allows us to customize things exactly how we'd like (regarding plugins).